src/mono/netcore/sample/iOS/Program.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+
+public class MonoPInvokeCallbackAttribute : Attribute
+{
+    public MonoPInvokeCallbackAttribute (Type delegateType) { }
+}
+
+public static class Program
+{
+    // Defined in main.m
+    [DllImport ("__Internal")]
+    private extern static void ios_set_text ([MarshalAs (UnmanagedType.LPUTF8Str)] string value);
+
+    [DllImport ("__Internal")]
+    private extern static void ios_register_button_click (Action action);
+
+    private static int counter = 0;
+
+    // Called by native code, see main.m
+    [MonoPInvokeCallback (typeof (Action))]
+    private static async void OnButtonClick ()
+    {
+        ios_set_text ("OnButtonClick! #" + counter++);
+    }
+
+    public static async Task Main (string[] args)
+    {
+        // Register a managed callback (will be called by UIButton, see main.m)
+        ios_register_button_click (OnButtonClick);
```

Comment:
The delegate created here doesn't need to be kept alive in some way? Or is it just used to get the function pointer?

Reply:
Good point! let me check. I usually just create static variables to keep callbacks alive always.